core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/sendtransaction/transaction/PrivateTransaction.java

```diff
@@ -0,0 +1,114 @@
+ */
+package tech.pegasys.web3signer.core.service.jsonrpc.handlers.sendtransaction.transaction;
+
+import tech.pegasys.web3signer.core.service.jsonrpc.EeaSendTransactionJsonParameters;
+import tech.pegasys.web3signer.core.service.jsonrpc.JsonRpcRequest;
+import tech.pegasys.web3signer.core.service.jsonrpc.JsonRpcRequestId;
+import tech.pegasys.web3signer.core.service.jsonrpc.handlers.sendtransaction.NonceProvider;
+
+import java.math.BigInteger;
+import java.util.List;
+
+import com.google.common.base.MoreObjects;
+import org.jetbrains.annotations.NotNull;
+import org.web3j.crypto.Sign.SignatureData;
+import org.web3j.protocol.eea.crypto.PrivateTransactionEncoder;
+import org.web3j.protocol.eea.crypto.RawPrivateTransaction;
+import org.web3j.rlp.RlpEncoder;
+import org.web3j.rlp.RlpList;
+import org.web3j.rlp.RlpType;
+
+public abstract class PrivateTransaction implements Transaction {
+
+  protected static final String JSON_RPC_METHOD = "eea_sendRawTransaction";
+  protected final long chainId;
+  protected final JsonRpcRequestId id;
+  protected final NonceProvider nonceProvider;
+  protected BigInteger nonce;
+  protected final EeaSendTransactionJsonParameters transactionJsonParameters;
+
+  PrivateTransaction(
+      final long chainId,
+      final EeaSendTransactionJsonParameters transactionJsonParameters,
+      final NonceProvider nonceProvider,
+      final JsonRpcRequestId id) {
+    this.chainId = chainId;
+    this.transactionJsonParameters = transactionJsonParameters;
+    this.nonceProvider = nonceProvider;
+    this.id = id;
+    this.nonce = transactionJsonParameters.nonce().orElse(null);
+  }
+
+  @Override
+  public void updateFieldsIfRequired() {
+    if (!this.isNonceUserSpecified()) {
+      this.nonce = nonceProvider.getNonce();
+    }
+  }
+
+  @Override
+  public byte[] rlpEncode(final SignatureData signatureData) {
+    final RawPrivateTransaction rawTransaction = createTransaction();
+    final List<RlpType> values =
+        PrivateTransactionEncoder.asRlpValues(rawTransaction, signatureData);
+    final RlpList rlpList = new RlpList(values);
+    return RlpEncoder.encode(rlpList);
+  }
+
+  @Override
+  public boolean isNonceUserSpecified() {
+    return transactionJsonParameters.nonce().isPresent();
+  }
+
+  @Override
+  public String sender() {
+    return transactionJsonParameters.sender();
+  }
+
+  @Override
+  public JsonRpcRequest jsonRpcRequest(
+      final String signedTransactionHexString, final JsonRpcRequestId id) {
+    return Transaction.jsonRpcRequest(signedTransactionHexString, id, getJsonRpcMethodName());
+  }
+
+  @Override
+  @NotNull
+  public String getJsonRpcMethodName() {
+    return JSON_RPC_METHOD;
+  }
+
+  @Override
+  public JsonRpcRequestId getId() {
+    return id;
+  }
+
+  @Override
+  public boolean isEip1559() {
+    return transactionJsonParameters.maxPriorityFeePerGas().isPresent()
+        && transactionJsonParameters.maxFeePerGas().isPresent();
+  }
```

Comment:
Thought it made sense to implement this (and the JsonParameters) since we might still want to act on eip1559 transactions even if we don't support them, for example producing a warning that is isn't supported for private txs yet